algosdk/encoding.py

```diff
@@ -103,6 +109,11 @@ def msgpack_decode(enc):
         Transaction, SignedTransaction, Multisig, Bid, or SignedBid:\
             decoded object
     """
+    warnings.warn(
+        "`msgpack_decode` is being deprecated. "
+        "Please use `future_msgpack_decode` instead.",
+        DeprecationWarning,
+    )
```

Comment:
I don't have much objection here, just a minor question: since encode-decode is coupled, do we want to mark encode as deprecated? Or it is because of `Transaction` had a structural change (i.e., some new fields)?

Reply:
Serializing into `msgpack` is okay - the old `Transaction` and new `Transaction` can both be serialized into msgpack as long as the fields are ordered and can be represented by a key-value map/dict. Decoding msgpack into the old `Transaction` does not seem preferable, so marked it with the deprecation warning here.